### ISSUE TITLE
Partial table cell and border alignment when content size is not known in advance

### ISSUE THREAD

**Issue by paplorinc** [flutter/flutter#107917]
## Steps to Reproduce

I am using table borders to draw double elimination brackets, similar ones to e.g. https://www.printyourbrackets.com/pdfbrackets/6teamDouble.pdf

To be able to add partial borders (not for entire rows or columns), I'm using the `decoration` of `Container`.
For some reason I can't seem to align the bordes properly, please see the example below:

<img width="998" alt="Screenshot 2022-07-19 at AM 11 33 03" src="https://user-images.githubusercontent.com/1841944/179719558-e6b6e8bb-4ef8-4401-a71c-17a347cf759e.png">

The desired outcome would be the last example, but with thicker borders (like in the first one).
Since I don't know in advance which cell will be the largest and can't apply `TableCellVerticalAlignment.fill` to every cell, is there a workaround I can try?

<details>
<summary>Bracket misalignment example</summary>

```dart
// https://dartpad.dev/?id=33b4e161d91df028d835c0795467e07f

import 'package:flutter/material.dart';

void main() => runApp(const MyApp());

class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) => const MaterialApp(home: Scaffold(body: MyStatelessWidget()));
}

class MyStatelessWidget extends StatelessWidget {
  const MyStatelessWidget({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return Column(children: [
      const Text("Unaltered"),
      createTable(),
      const Text("withNoBorderWidth"),
      createTable(withNoBorderWidth: true),
      const Text("withFill"),
      createTable(withFill: true),
      const Text("withNoBorderWidth & withFill"),
      createTable(withNoBorderWidth: true, withFill: true),
    ]);
  }

  Container createTable({bool withNoBorderWidth = false, bool withFill = false}) {
    var cellBackgroundColor = Colors.red.withAlpha(50),
        borderSide = BorderSide(width: withNoBorderWidth ? 0 : 5), // FIXME width of 0 eliminates some of the gap
        topRightBorder = BoxDecoration(border: Border(top: borderSide, right: borderSide), color: cellBackgroundColor),
        bottomRightBorder = BoxDecoration(border: Border(right: borderSide, bottom: borderSide), color: cellBackgroundColor),
        rightBorder = BoxDecoration(border: Border(right: borderSide), color: cellBackgroundColor),
        bottomBorder = BoxDecoration(border: Border(bottom: borderSide), color: cellBackgroundColor);

    var emptyCell = const MapEntry(BoxDecoration(), Text(""));

    var rows = [
      [MapEntry(topRightBorder, const Text("A")), emptyCell, emptyCell],
      [MapEntry(bottomRightBorder, const Text("B", style: TextStyle(fontSize: 30))), MapEntry(topRightBorder, const Text("E")), emptyCell],
      [emptyCell, MapEntry(rightBorder, const Text("")), MapEntry(bottomBorder, const Text("G"))],
      [emptyCell, MapEntry(bottomRightBorder, const Text("F")), emptyCell],
    ].map((r) => r.map((c) => Container(decoration: c.key, child: c.value) as Widget).toList()).toList();

    if (withFill) {
      // FIXME don't know in advance which cell will be the largest and can't apply to every cell
      rows[1][1] = TableCell(verticalAlignment: TableCellVerticalAlignment.fill, child: rows[1][1]);
    }
    return Container(
      margin: const EdgeInsets.all(20),
      child: Table(
        defaultVerticalAlignment: TableCellVerticalAlignment.middle,
        children: rows.map((r) => TableRow(children: r)).toList(),
      ),
    );
  }
}
```
</details>

**Comment by huycozy** [comment]
Hi @paplorinc 
It seems you are asking for assistance on personal code. Please see https://flutter.dev/community for resources and ask questions like this.
You may also get some help if you post it on Stack Overflow and if you need help with your code, please see https://www.reddit.com/r/flutterhelp.
Closing, as this isn't an issue with Flutter itself. If you disagree, please write in the comments and I will reopen it.

Thank you!

**Comment by paplorinc** [comment]
Hey @huycozy, thanks for your response.

I have already "workarounded" this in my code, so I thought I'll open a detailed issue describing it to either decide if this is a bug (my reference was the behavior of Google Spreadsheet or plain HTML where I could set the borders for cells instead of rows and columns and where it was trivial to create these brackets) or a feature request - or maybe a way to help others work around these limitations (I saw that others have had difficulties with similar requests on StackOverflow as well).

If you agree, that - based on the above code screenshot - it's a bug, please reopen.

**Comment by huycozy** [comment]
Hi @paplorinc 

> Since I don't know in advance which cell will be the largest and can't apply TableCellVerticalAlignment.fill to every cell, is there a workaround I can try?

Based on your question, it seems to ask for assistance for your project than reporting a bug.
The goal of this Github repo is not to give assistance on personal code or the whole package/plugin project,
but to improve flutter reporting bugs or with feature requests.

If you encounter behavior that you believe could be a bug, please describe the issue and indicate it is a bug that belongs to Flutter. And please try to provide all or some of the information indicated in the template so that we can address this issue properly.

<details>
<summary>Issue Template</summary>

```
If you are looking for support, please check out our documentation
or consider asking a question on Stack Overflow:
- https://flutter.dev/
- https://api.flutter.dev/
- https://stackoverflow.com/questions/tagged/flutter?sort=frequent

If you have found a bug or if our documentation doesn't have an answer
to what you're looking for, then fill our the template below. Please read
our guide to filing a bug first: https://flutter.dev/docs/resources/bug-reports

Steps to Reproduce

Please tell us exactly how to reproduce the problem you are running into.

Please attach a small application (ideally just one main.dart file) that
reproduces the problem. You could use https://gist.github.com/ for this.

If the problem is with your application's rendering, then please attach
a screenshot and explain what the problem is.

1. ...
2. ...
3. ...


Please tell us which target platform(s) the problem occurs (Android / iOS / Web / macOS / Linux / Windows)
Which target OS version, for Web, browser, is the test system running?
Does the problem occur on emulator/simulator as well as on physical devices?

Target Platform:
Target OS version/browser:
Devices:

Run your application with `flutter run --verbose` and attach all the
log output below between the lines with the backticks. If there is an
exception, please see if the error message includes enough information
to explain how to solve the issue.

Run `flutter analyze` and attach any output of that command below.
If there are any analysis errors, try resolving them before filing this issue.

Finally, paste the output of running `flutter doctor -v` here.
```
</details>

Instead, if you think that the issue might depend on your implementation, please see flutter's [community](https://flutter.dev/community) for resources and ask questions like this. You may also get some help if you post it on Stack Overflow and [flutterhelp](https://www.reddit.com/r/flutterhelp/).

Thank you!

**Comment by paplorinc** [comment]
> it seems to ask for assistance for your project than reporting a bug

I don't need assistance, I already found a workaround, as stated previously.

I've opened the report because the behavior of the Table widget seems inconsistent with other Table implementations we may be used to, for example creating the above in Google Spreadsheets looks as intended without any workarounds:
<img width="551" alt="image" src="https://user-images.githubusercontent.com/1841944/180163886-df087db1-6783-4c61-9f90-e732ed4f6439.png">

The same with Flutter Table looks incomplete:
<img width="802" alt="image" src="https://user-images.githubusercontent.com/1841944/180164131-734bab6d-59ef-4189-a40c-f62a30834705.png">

Do you need more information or is this enough for you to investigate?

**Comment by huycozy** [comment]
Hi @paplorinc, I reopen this issue for further discussion.

Please describe `Actual Results` and `Expected results` clearly so that we may verify this.

> Since I don't know in advance which cell will be the largest and can't apply TableCellVerticalAlignment.fill to every cell, is there a workaround I can try?

> I have already "workarounded" this in my code, so I thought I'll open a detailed issue describing it to either decide if this is a bug (my reference was the behavior of Google Spreadsheet or plain HTML where I could set the borders for cells instead of rows and columns and where it was trivial to create these brackets) or a feature request 

You're asking for a workaround whilst you have already "workarounded". Or perhaps you will provide a workaround then, won't you?

> the behavior of the Table widget seems inconsistent with other Table implementations we may be used to

Please help to make this clear, how is it inconsistent?

**Comment by paplorinc** [comment]
> You're asking for a workaround whilst you have already "workarounded". Or perhaps you will provide a workaround then, won't you?

Yes, the workarounds are in the code and the screenshots show the differences:
`Container createTable({bool withNoBorderWidth = false, bool withFill = false}) {`
I have hard-coded the `TableCellVerticalAlignment.fill` cells in my table.

> Please help to make this clear, how is it inconsistent?

1) Changing the border width introduces offsets to the other rows:
<img width="796" alt="image" src="https://user-images.githubusercontent.com/1841944/180416379-478b2cfc-2af6-4b39-98fe-7491f64b11fa.png">

2) There no way currently (or at least I couldn't find one, hence my question for better workaround) to make the table just auto-fill without me having to select some special cells that should expand and which shouldn't, as shown in https://github.com/flutter/flutter/issues/107917#issuecomment-1191182066

# Actual Results

The table cells need manual adjustments to not have holes in them.
 
# Expected results

I create the rows and columns and the table calculates the final cell sizes automatically without me having to know the layout of the cells in advance - similarly to HTML tables and how Google Spreadsheets does it.

**Comment by huycozy** [comment]
Hi @paplorinc 
> There no way currently (or at least I couldn't find one, hence my question for better workaround) to make the table just auto-fill without me having to select some special cells that should expand and which shouldn't

Thank you for updating the issue. Labeling this issue as a proposal for further opinions from the team!
